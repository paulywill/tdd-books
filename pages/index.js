import Head from 'next/head'
import Link from 'next/link'
import Layout, { siteTitle } from '../components/layout'
import utilStyles from '../styles/utils.module.css'


export default function Home() {
  return (
    <Layout home>
      <Head>
        <title>{siteTitle}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <section className={utilStyles.headingMd}>
        <p>This is a TDD Book Catalog project. I'm hoping it will allow me to refresh my learning of React and also learn more about Nextjs, Typescript, and Jest.</p>
      </section>
   </Layout>
  )
}