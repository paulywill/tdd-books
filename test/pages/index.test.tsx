import React from 'react'
import { render, fireEvent, screen } from '../testUtils'
import { Home } from '../../pages/index'
import '@testing-library/jest-dom/extend-expect'

//describe('App', () => {
//test('renders App component', () => {
//render(<Home />);
//screen.debug();
//});
//});

describe('Home page', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  test('title is correct', () => {
    render(<Home />)
    expect(screen.getByText(/Welcome/)).toBeInTheDocument()
    expect(screen.getByRole('heading')).toBeInTheDocument()
  })

  test('clicking button triggers alert', () => {
    const { getByText } = render(<Home />, {})
    window.alert = jest.fn()
    fireEvent.click(getByText('Test Button'))
    expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  })
})
