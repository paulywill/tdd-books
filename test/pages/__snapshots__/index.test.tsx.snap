// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-2250570153 container"
  >
    <main
      class="jsx-2250570153"
    >
      <h1
        class="jsx-2250570153 title"
      >
        Welcome to TDD Books!
      </h1>
      <p
        class="jsx-2250570153 description"
      />
      <button
        class="jsx-2250570153"
      >
        Test Button
      </button>
    </main>
    <footer
      class="jsx-2250570153"
    >
      Powered by â˜• , ðŸŽ§, , ðŸ¤“
    </footer>
  </div>
</DocumentFragment>
`;
